src/main/java/org/openrewrite/staticanalysis/MethodNameCasing.java

```diff
@@ -110,40 +100,10 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, Ex
                 if (containsValidModifiers(method) &&
                     method.getMethodType() != null &&
                     enclosingClass.getType() != null &&
-                    !method.isConstructor() &&
-                    !simpleName.startsWith("_") &&
-                    !STANDARD_METHOD_NAME.matcher(simpleName).matches()) {
-                    StringBuilder standardized = new StringBuilder();
+                    !method.isConstructor()) {
                     String normalized = VariableNameUtils.normalizeName(simpleName);
-
-                    if (SNAKE_CASE.matcher(normalized).matches()) {
-                        standardized.append(NameCaseConvention.format(NameCaseConvention.LOWER_CAMEL, normalized));
-                    } else {
-                        int nameLength = normalized.length();
-                        for (int i = 0; i < nameLength; i++) {
-                            char c = normalized.charAt(i);
-
-                            if (i == 0) {
-                                // the java specification requires identifiers to start with [a-zA-Z$_]
-                                if (c != '$' && c != '_') {
-                                    standardized.append(Character.toLowerCase(c));
-                                }
-                            } else {
-                                if (!Character.isLetterOrDigit(c)) {
-                                    while (i < nameLength && (!Character.isLetterOrDigit(c) || c > 'z')) {
-                                        c = normalized.charAt(i++);
-                                    }
-                                    if (i < nameLength) {
-                                        standardized.append(Character.toUpperCase(c));
-                                    }
-                                } else {
-                                    standardized.append(c);
-                                }
-                            }
-                        }
-                    }
-
-                    String toName = standardized.toString();
+                    NamingService service = service(NamingService.class);
+                    String toName = service.standardizeMethodName(normalized);
                     if (!StringUtils.isBlank(toName) && !StringUtils.isNumeric(toName) &&
                         !methodExists(method.getMethodType(), toName)) {
```

Comment:
`standardizeMethodName` is not guaranteed to give back a different method name; do we already have a test to check expecting no changes? Or should we add a comparison here between the old and new name?

Reply:
Think it's best to introduce a check here, better safe then sorry